addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/handler/YamahaZoneThingHandler.java

```diff
@@ -0,0 +1,606 @@
+            }
+        }
+
+        if (includeInputWithNavigationControl) {
+            logger.debug("navigation control requested by channel");
+            if (InputWithNavigationControl.supportedInputs.contains(currentInputID)) {
+                inputWithNavigationControl = new InputWithNavigationControl(currentInputID,
+                        brHandler.getCommunication(), this);
+                try {
+                    inputWithNavigationControl.updateNavigationState();
+                } catch (Exception e) {
+                    logger.error(e.getMessage());
+                }
+            } else {
+                logger.debug("navigation control  not supported by " + currentInputID);
+                inputWithNavigationControl = null;
+                naviState.invalidate();
+                navigationUpdated(naviState);
+            }
+        } else {
+            inputWithNavigationControl = null;
+            naviState.invalidate();
+            navigationUpdated(naviState);
+        }
+
+        boolean includeInputWithPlaybackControl = false;
+
+        for (String channelName : YamahaReceiverBindingConstants.CHANNELS_PLAYBACK) {
+            if (isLinked(grpPlayback(channelName))) {
+                includeInputWithPlaybackControl = true;
+                break;
+            }
+        }
+
+        if (includeInputWithPlaybackControl) {
+            logger.debug("playback control requested by channel");
+            if (InputWithPlayControl.supportedInputs.contains(currentInputID)) {
+                inputWithPlayControl = new InputWithPlayControl(currentInputID, brHandler.getCommunication(), this);
+                try {
+                    inputWithPlayControl.updatePlaybackInformation();
+                    inputWithPlayControl.updatePresetInformation();
+                } catch (Exception e) {
+                    logger.error(e.getMessage());
+                }
+            } else {
+                logger.debug("playback control  not supported by " + currentInputID);
+                inputWithPlayControl = null;
+                playControlState.invalidate();
+                playInfoState.invalidate();
+                playControlUpdated(playControlState);
+                playInfoUpdated(playInfoState);
+            }
+        } else {
+            inputWithPlayControl = null;
+            playControlState.invalidate();
+            playControlUpdated(playControlState);
+            playInfoUpdated(playInfoState);
+            playInfoState.invalidate();
+        }
+
+    }
+
+    /**
+     * Once this thing is set up and the AVR is connected, the available inputs for this zone are requested.
+     * The thing is updated with a new CHANNEL_INPUT which only lists the available zones instead of all known zones.
+     */
+    @Override
+    public void availableInputsChanged(AvailableInputState msg) {
+        // ChannelUID channelUID = new ChannelUID(thing.getUID(), YamahaReceiverBindingConstants.CHANNEL_GROUP_ZONE,
```

Comment:
This is a todo and has to be done by another merge request. But it took a day to get this far, and I don't really want to remove this commented code right now, because it is partly doing what I wanted to archive.
Is there a developer documentation of how to dynamically change channel states that are statically defined in the xml?